hpke/algs.go

```diff
@@ -44,7 +49,14 @@ func (k KEM) IsValid() bool {
 	}
 }
 
-func (k KEM) Scheme() kem.AuthScheme {
+// AuthScheme returns an instance of a KEM that supports authentication. If
+// the KEM does not support authentication, it returns nil. Panics if the KEM
+// identifier is not valid.
+func (k KEM) AuthScheme() kem.AuthScheme { return k.Scheme().(kem.AuthScheme) }
```

Comment:
This seems like option (2.) from https://github.com/armfazh/circl/pull/8#discussion_r542501005.

I still don't think this is a good idea. The complexity it adds to the API isn't needed, since we currently have no HPKE KEMs that aren't suitable for sender authentication. I vote we hold off on this change until we actually need it. At the very least, let's rever this change in the current PR, since it's not relevant to the API changes for KDF. We can discuss it later on.

Reply:
reverted